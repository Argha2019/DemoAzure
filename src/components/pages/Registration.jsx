import React from 'react';
import HomePageTemplate from "../templates/HomePageTemplate/Template.jsx";
import AgentTemplate from "../templates/AgentTemplate/Template.jsx";
import AdminTemplate from "../templates/AdminTemplate/Template.jsx";
import Cookies from 'js-cookie';
import RegistrationTemplate from '../templates/RegistrationTemplate/Template.jsx';


class Registration extends React.Component {

	render() {

		if (Cookies.get('adminData')) {
			return (
				<AdminTemplate>
					<RegistrationTemplate isAdmin />
				</AdminTemplate>
			);
		}

		else if (Cookies.get('agentData')) {
			return (
				<AgentTemplate>
					<RegistrationTemplate />
				</AgentTemplate>
			);
		}

		else {
			return (
				<HomePageTemplate>
					<RegistrationTemplate />
				</HomePageTemplate>
			)
		}
	}
}

export default Registration;
