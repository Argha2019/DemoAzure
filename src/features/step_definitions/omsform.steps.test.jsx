import React from 'react';
import { shallow, mount } from 'enzyme';
import OMSForm from '../../components/pages/OmsForm.jsx';
import toJson from 'enzyme-to-json';
import Registration from '../../components/pages/Registration.jsx';
import { defineFeature, loadFeature } from 'jest-cucumber';

const feature = loadFeature('./src/features/omsform.feature');

defineFeature(feature, test => {
	test('Rendering Snapshot', ({given,when,then}) => {

		let wrapper;

		given('snapshot given', () => {

		});

		when('mounting snapshot', () => {
			wrapper = shallow(<OMSForm />);
		});

		then('it should render the snapshot', () => {
			expect(toJson(wrapper)).toMatchSnapshot();
		});

	});

	test('Should render required form elements', ({given,when,then}) => {

		let wrapper;
		let form;

		given('snapshot given', () => {

		});

		when('finding Form', () => {
			wrapper = shallow(<OMSForm />);
			form = wrapper.find('Form');
		});

		then('it should find FormControl And Buttons', () => {
			expect(form.length).toBe(1);
	   		expect(form.find('Button').length).toBe(1);
	   		expect(form.find('Button').props().disabled).toBe(false);
	     	expect(form.find('FormControl').length).toBe(2);
		});

	});

	test('Should render the OMSForm and find div and buttons', ({given,when,then}) => {

		let wrapper;
		let oms;

		given('snapshot given', () => {

		});

		when('finding div and buttons', () => {
			wrapper = shallow(<OMSForm />);
			oms = wrapper.find('div');
		});

		then('it should get 1 div and 2 buttons', () => {
			expect(oms.length).toBe(3);
   			expect(oms.find('Button').length).toBe(2);
		});

	});

	test('Register Button Definition', ({given,when,then}) => {

		let register;
		let onRegister;
		let button;

		given('snapshot given', () => {

		});

		when('finding button register', () => {
			onRegister = jest.fn();
    		register = shallow(<OMSForm onRegister={onRegister} />);
    		button = register.find('.register').first();
		});

		then('it should be defined', () => {
			expect(button).toBeDefined();
		});

	});

	test('OnChange function', ({given,when,then}) => {

		let event;
		let wrapper;
		let input;

		given('snapshot given', () => {

		});

		when('changing values in fields', () => {
			event = {
		      preventDefault: () => {},
		      target: {
		          name: "username",
		          value: "hasan",
		        }
      		}

	    	wrapper = shallow(<OMSForm />);

	    	wrapper.setState({
	      		username: "hasan",
	      		password: "hasan"
	    	});

	    	input = wrapper.find('FormControl').first();
	    	input.simulate('change', event);
	    	wrapper.update();
			});

		then('onChange should be called and state should be updated', () => {
			expect(event.target.value).toBe('hasan');
    		expect(event.target.name).toBe("username");
		});

	});

	test('on Button Click', ({given,when,then}) => {

		let event;
		let log;
		let wrapper;
		let form;

		given('snapshot given', () => {

		});

		when('submit button is clicked', () => {
			event = {
      			preventDefault: () => {},
    		}

    		log = jest.fn();
    		global.console = {log};

    		wrapper = shallow(<OMSForm state={{username: "hasan"}} onSubmit={log} />);
    		form = wrapper.find('Form');
    		form.simulate('submit', event);
		});

		then('check valid credentials', () => {
			expect(log).toHaveBeenCalledTimes(0);
    		expect(wrapper.state('validCredential')).toBe('');
		});

	});
});
