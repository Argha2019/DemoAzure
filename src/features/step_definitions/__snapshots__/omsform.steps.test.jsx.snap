// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OMSForm Testing Rendering Snapshot 1`] = `
<HomePageTemplate
  className=""
>
  <div>
    <img
      alt="e-commerce image"
      className="loginImage"
      height="800px"
      src="https://psi-oms-product-image-repo.s3.us-east-2.amazonaws.com/omslogin.png"
      width="70%"
    />
    <div
      align="center"
      className="login-wrap"
    >
      <div
        className="login-html"
      >
        <h1
          className="fontWhite"
        >
          Login
        </h1>
        <Form
          className="login-form MyForm"
          inline={false}
          onSubmit={[Function]}
        >
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="username"
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Username
            </FormLabel>
            <FormControl
              className="input"
              name="username"
              onChange={[Function]}
              placeholder="Username"
              required={true}
              type="text"
            />
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="password"
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Password
            </FormLabel>
            <FormControl
              className="input"
              name="password"
              onChange={[Function]}
              placeholder="password"
              required={true}
              type="password"
            />
          </FormGroup>
          <br />
          <Button
            active={false}
            className="button"
            disabled={false}
            type="submit"
            variant="primary"
          >
            Sign In
          </Button>
        </Form>
        <p
          className="fontWhite"
        >
          New User? 
        </p>
        <Button
          active={false}
          className="registerButton"
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Register
        </Button>
      </div>
    </div>
  </div>
</HomePageTemplate>
`;
